src/components/dbCalls.js

```diff
@@ -0,0 +1,435 @@
+        const taskQuery = query(taskCollectionRef, where ('isComplete', '==', true));
+        const taskSnapshot = await getDocs(taskQuery);
+        return taskSnapshot;
+    }
+    catch (error) { 
+        console.error('Error fetching inactive tasks: ', error);
+    }
+}
+
+//fetches a task's data from database
+const fetchTaskForRoom = async (taskID, roomID) => {
+    try {
+        const taskCollectionRef = collection(db, 'rooms', roomID, 'tasks');
+        const taskRef = doc(taskCollectionRef, taskID);
+        const taskSnapshot = await getDoc(taskRef);
+        return taskSnapshot.data();
+    }
+    catch (error) {
+        console.error('Error fetching task: ', error);
+    }
+}
+
+//updates player's score
+const updatePointsForPlayer = async (player, points, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', player));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerdoc = playerSnapshot.docs[0].ref;
+        const playerScore = parseInt(playerSnapshot.docs[0].data().score);
+        const newPoints = points + playerScore;
+        await updateDoc(playerdoc, { score: newPoints });
+    }
+    catch (error) {
+        console.error('Error updating player score: ', error);
+    }
+}
+
+//revives a player
+const updateIsAliveToTrueForPlayer = async (player, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', player));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerdoc = playerSnapshot.docs[0].ref;
+        await updateDoc(playerdoc, { isAlive: true });
+    }
+    catch (error) {
+        console.error('Error reviving player: ', error);
+    }
+}
+
+//turns a player's alive status to false
+const updateIsAliveToFalseForPlayer = async (player, roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerQuery = query(playerCollectionRef, where('name', '==', player));
+        const playerSnapshot = await getDocs(playerQuery);
+        const playerdoc = playerSnapshot.docs[0].ref;
+        await updateDoc(playerdoc, { isAlive: false });
+    }
+    catch (error) {
+        console.error('Error killing player: ', error);
+    }
+}
+
+//marks task as completed
+const updateIsCompleteToTrueForTask = async (taskID, roomID) => {
+    try {
+        const taskCollectionRef = collection(db, 'rooms', roomID, 'tasks');
+        const taskDocRef = doc(taskCollectionRef, taskID);
+        await updateDoc(taskDocRef, { isComplete: true });
+    }
+    catch (error) {
+        console.error('Error completing task: ', error);
+    }
+}
+
+//updates the 'completedBy' field of a task
+const updateCompletedByForTask = async (taskDocRef, checkedPlayers) => {
```

Comment:
Hmm... is it necessary for this to be a separate method from the above method? I can't think of a time when we would want to change the completedBy field without marking the task completed, or vice versa. Is there such a time?

If not, you probably want to rewrite things a bit so that both these updates occur at the same time. It makes your life easier (you don't have to pass a taskDocRef) and a little more resilient to errors (you wouldn't want some weird error happening in between updating the task to completed and updating who completed it, so that)

Reply:
updating the 'completedBy' field is important because it keeps track of who completed the task whenever the selected task is switched or page is refreshed. If changes were saved, and either situation mentioned were to occur, a useEffect hook is triggered to look at the 'completedBy' field to recheck everyone that had been saved. Do you think there is a better way to do this?

Reply:
Ahh right, I gotcha, I forgot how the UI flow of this thing works. Ok, that's fine for now then, ignore this comment